homeassistant/components/instructure/sensor.py

```diff
@@ -213,31 +257,36 @@ async def async_setup_entry(
     """Set up Canvas sensor based on a config entry."""
     coordinator = hass.data[DOMAIN][entry.entry_id]["coordinator"]
 
-    def update_entities(data_type: str, new_data: dict, curr_data: dict):
-        """Add or remove sensor entities based on new data"""
-        current_ids = set(curr_data.keys())
-        new_ids = set(new_data.keys())
-
-        to_add = new_ids - current_ids
-        to_remove = current_ids - new_ids
-
-        for entity_id in to_remove:
-            entity = hass.data[DOMAIN][entry.entry_id]["entities"][data_type][entity_id]
-            entity.async_remove(force_remove=True)
-
+    def add_new_entities(data_type: str, entity_ids: [str]) -> None:
         new_entities = []
-
-        for entity_id in to_add:
+        for entity_id in entity_ids:
             description = SENSOR_DESCRIPTIONS[data_type]
             new_entity = CanvasSensorEntity(description, entity_id, coordinator)
             new_entities.append(new_entity)
 
-            hass.data[DOMAIN][entry.entry_id]["entities"][data_type][
-                entity_id
-            ] = new_entity
+            hass.data[DOMAIN][entry.entry_id]["entities"][data_type][entity_id] = new_entity
 
         if new_entities:
             async_add_entities(tuple(new_entities))
 
+    def update_entities(new_data: dict[str, Any]):
+        """Add or remove sensor entities based on new data"""
+        registry = er.async_get(hass)
+        existing_sensors = {entity_id: entry for entity_id, entry in registry.entities.items() if entry.platform == 'instructure' and entity_id.split(".")[0]=="sensor"}
+        existing_sensor_ids = set([entity_id for entity_id, _ in existing_sensors.items()])
+
+        all_new_entity_ids = []
```

Comment:
why not make this a set also?

Reply:
converted it to a set in line 286 but you're right. It's better to just create it as a set. Will do